### PR TITLE
Add `DensifyGeodesic` trait

### PR DESCRIPTION
**Pull request by grevgeny**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

This PR adds `DensifyGeodesic` trait that follows logic similar to `DensifyHaversine`. However, the trait is only implemented for `T = f64` as underlying Geodesic traits are implemented only for this type.

### CODE REVIEW

File: geo/src/algorithm/densify_geodesic.rs
Comment (michaelkirk) [inline geo/src/algorithm/densify_geodesic.rs:34]:
```suggestion
    fn densify_geodesic(&self, max_meters: f64) -> Self::Output {
```

I know it's in the docs, but it's nice to be redundant sometimes.

File: geo/src/algorithm/densify_geodesic.rs
Comment (michaelkirk) [inline geo/src/algorithm/densify_geodesic.rs:167]: Can you add a test where `max_meters` is larger than each of the segment lengths?   i.e. where the algorithm is a no-op.

File: geo/src/algorithm/mod.rs
Comment (michaelkirk) [inline geo/src/algorithm/mod.rs:85]:
```suggestion
/// Densify geometry components on a geodesic.
```

nit: period, not semicolon.

And I don't know that it's better, but "geodesic"  is consistent with our other doc strings, so let's use it everywhere.

**Review by michaelkirk — CHANGES_REQUESTED** [review]
I have some minor nits, but overall it looks good. Thank you!

**Comment by grevgeny** [comment]
@michaelkirk thanks for your review! I have added proposed changes

**Comment by michaelkirk** [comment]
I think you'll need to run cargo fmt, otherwise LGTM

I'll leave it open for a couple days before merging in case anyone else wants to weigh in.

**Comment by michaelkirk** [comment]
Darn it. I'm sorry for not realizing this earlier. This seems related to https://github.com/georust/geo/issues/1181

**Comment by michaelkirk** [comment]
Assuming #1216 gets merged, I've got a different approach that should supersede this, generically implementing Densify for all line measures (Haversine, Geodesic, Euclidean, etc.)

You can see a WIP here: https://github.com/georust/geo/compare/mkirk/unify-densify?expand=1

**Comment by michaelkirk** [comment]
My alternative to this was merged in #1231.

With that you can call:
```rust
use geo::{Geodesic, Densify, Euclidean};
line_string.densify::<Geodesic>();
line_string.densify::< Euclidean >();
```

Let me know if that works for your purposes @grevgeny.

**Comment by michaelkirk** [comment]
With #1298 (not yet released) this changed to:

```rust
use geo::{Geodesic, Densify, Euclidean};
Geodesic.densify(&line_string);
Euclidean.densify(&line_string);
```

**Pull request closed without merging**
